### PR TITLE
[ADD] add new addon to allow searches based on a list of values

### PR DESCRIPTION
**Pull request by Cedric-Pigeon**

### CODE REVIEW

**Comment by rousseldenis** [comment]
IMHO, this should be moved to OCA/product-attribute. No ?

File: product_search_multi_value/models/product_template.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        default_res = FALSE_DOMAIN
+        if operator in ["=", "ilike"]:
+            operator = "in"
+            comparator = OR
+        elif operator in ["!=", "not ilike"]:
+            operator = "not in"
+            comparator = AND
+        else:
+            return default_res
+        value_list = value.split(" ") if " " in value else [value]
```
Comment (sebastienbeau) [inline product_search_multi_value/models/product_template.py:32]: does the separator should be configurable (with a ir.config_parameter) ?

**Comment by sebastienbeau** [comment]
An other implementation possibility will to have a option on the ir.fields (multi-search) and inherit the generic search method of base class.
Then if you have tic the option on the default_code field of the product, this code will parse the domain to split the value passed.

The advantage is that we can have this feature on every object (search several PO, several SO ...)

File: product_search_multi_value/models/product_template.py (hunk 1 above)
Comment (Cedric-Pigeon) [reply product_search_multi_value/models/product_template.py:32]: not sure, the goal was to allow copy/paste for Excel cells without having to rework the result. and by default it set a space.

File: product_search_multi_value/data/search_field_data.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<odoo noupdate="1">
+    <record model="ir.config_parameter" id="search_fields">
+        <field name="key">product_search_multi_value.search_fields</field>
+        <field name="value">['default_code', 'barcode']</field>
```
Comment (rousseldenis) [inline product_search_multi_value/data/search_field_data.xml]: @Cedric-Pigeon barcode is a wrong example as is a related on product_variant_ids.barcode.

In v10, some product.product are active=False (with product.template active = True), so the generated query for 'not ilike' will exclude products (even templates) for which variants are deactivated.

As I don't see any mean to do that, I will do the tests on another field, set in README the limitation.

**Pull request closed without merging**